papermill/iorw.py

```diff
@@ -422,6 +424,22 @@ def load_notebook_node(notebook_path):
 
         if not hasattr(cell.metadata, 'papermill'):
             cell.metadata['papermill'] = dict()
+
+    # Force refreshing default parameters
+    parameter_cell_idx = find_first_tagged_cell_index(nb, "parameters")
+    parameter_cell = nb.cells[parameter_cell_idx]
+    kernel_name = nb.metadata.kernelspec.name
+    language = nb.metadata.kernelspec.language
+
+    translator = papermill_translators.find_translator(kernel_name, language)
+    try:
+        params = translator.inspect(parameter_cell)
+        nb.metadata['papermill']['default_parameters'] = {p.name: p._asdict() for p in params}
+    except NotImplementedError:
+        logger.warning(
```

Comment:
This might get annoying having this print on each notebook execution for language without inspection support. I think we should only pull out default parameters on demand when `--help` is inquired to avoid A) unnecessary warnings and B) errors in inspection blocking execution paths.

Reply:
The goal was to achieve the proposal of #225.

But I totally get your point.